miden-lib/src/transaction/inputs.rs

```diff
@@ -59,22 +63,26 @@ fn build_advice_stack(
     tx_script: Option<&TransactionScript>,
     inputs: &mut AdviceInputs,
 ) {
+    let header = tx_inputs.block_header();
+
+    // push the accumulative hash of all kernels to the advice stack
+    inputs.extend_stack(header.kernel_root());
```

Comment:
As mentioned in one of the previous comments - I think we can avoid this, but we'll need to put `kernel_offset` onto the advice stack.